packages/editor/src/components/block-bindings/bindings-ui.js

```diff
@@ -0,0 +1,278 @@
+																fill[ 0 ].props
+																	.children
+																	.props
+																	.source;
+															const sourceLabel =
+																fill[ 0 ].props
+																	.children
+																	.props
+																	.label;
+															const isSourceSelected =
+																activeSource ===
+																source;
+
+															return (
+																<Fragment
+																	key={
+																		index
+																	}
+																>
+																	<MenuItem
+																		icon={
+																			isSourceSelected
+																				? chevronUp
+																				: chevronDown
+																		}
+																		isSelected={
+																			isSourceSelected
+																		}
+																		onClick={ () =>
+																			setIsActiveSource(
+																				isSourceSelected
+																					? false
+																					: source
+																			)
+																		}
+																		className="block-bindings-source-picker-button"
+																	>
+																		{
+																			sourceLabel
+																		}
+																	</MenuItem>
+																	{ isSourceSelected &&
+																		fill }
+																</Fragment>
+															);
+														}
+													) }
+												</>
+											);
+										} }
+									</Slot>
+								</MenuGroup>
+								<RemoveBindingButton
+									{ ...props }
+									currentAttribute={ attribute }
+									setIsActiveAttribute={
+										setIsActiveAttribute
+									}
+								/>
+							</>
+						) }
+					</div>
+				) ) }
+			</MenuGroup>
+		);
+	}
+
+	function RemoveBindingButton( props ) {
+		return (
+			<Button
+				className="block-bindings-remove-button"
+				onClick={ () => {
+					if ( ! props.attributes?.metadata.bindings ) {
+						return;
+					}
+
+					const { currentAttribute } = props;
+					// Modify the attribute we are binding.
+					const newAttributes = {};
+					newAttributes[ currentAttribute ] = '';
+					setAttributes( newAttributes );
+
+					// Update the bindings property.
+					updateBlockBindingsAttribute(
+						props.attributes,
+						setAttributes,
+						currentAttribute,
+						null
+					);
+				} }
+			>
+				Remove binding
+			</Button>
+		);
+	}
+
+	const [ popoverAnchor, setPopoverAnchor ] = useState();
+	return (
+		<>
+			<BlockControls group="other">
+				<Button
+					onClick={ () => {
+						setAddingBinding( ! addingBinding );
+					} }
+					aria-expanded={ true }
+					icon={ pluginsIcon }
+					ref={ setPopoverAnchor }
+				></Button>
+				{ addingBinding && <BindingsUI /> }
+			</BlockControls>
+		</>
+	);
+};
+
+if ( window.__experimentalBlockBindings ) {
+	addFilter(
+		'blocks.registerBlockType',
+		'core/block-bindings-ui',
+		( settings, name ) => {
+			if ( ! ( name in blockBindingsWhitelist ) ) {
+				return settings;
+			}
+
+			// TODO: Review the implications of this and the code.
```

Comment:
Can you explain what this code is doing? If the idea is to expose the context to all blocks that allow block bindings, then maybe it would work best to make these context entries global, or on the server they should be automatically injected during the block registration when the block allows binding.

Reply:
There are two parts:
* Exposing the context to all blocks.
* Injecting the `<BlockBindingsUI />` component.

Regarding adding the context to all blocks I am not 100% sure if it is needed. Maybe sources like [this one](https://github.com/WordPress/gutenberg/blob/c3f709a329dba397e46f78ce784eeed3017dc989/packages/editor/src/hooks/block-bindings-sources/post-meta.js), can get the `postType` and `postID` without needed to add it to the `block.json`. Any ideas if that's possible?

Reply:
I changed the implementation with this idea in this commit of another pull request (I've split this prototype in smaller PRs): [link](https://github.com/WordPress/gutenberg/pull/57258/commits/d43ed33ae8a2f69e085f87549623dbf185c38092).

In the post meta source, I'm getting the `postType` and the `postId` using `select( editorStore ).getCurrentPostId()` and `select( editorStore ).getCurrentPostType()` when they are not defined. That way, we can decide to not show that source if we are in a taxonomy for example. And we don't have to register the context as we were doing.

Let me know what you think and I can reuse it in the rest of the sources 🙂

Reply:
Nevermind, I believe in that approach, we're not getting the context properly when we are in a query loop, for example.